packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -585,13 +586,16 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
     /**
      * Closes the connection and clears inbound & outbound queues.
      */
-    public close(error?: ICriticalContainerError): void {
-        if (this.closed) {
+    public close(error?: ICriticalContainerError, emitDisposed?: boolean): void {
```

Comment:
subjective nit: I'd put the default value here rather than letting it be undefined. It's more self-documenting to illustrate what the default is.

Reply:
And also I'd suggest renaming this to `shouldDispose` or something, since it's not just about emitting an event (i.e. `disposeInternal` also removes listeners and could be updated to do other stuff later too)

Reply:
Oh... I see now that if `emitDisposed` is true then `closed` event is not emitted... That's surprising to me at this point in looking at the PR.  I'll come back here if I have more thoughts after seeing more.

Reply:
Ok I would suggest exposing `public dispose(...)` on this class.  `Container` can access it because it uses the class directly not via interface.  Then It will be more symmetrical / clear about the two codepaths (close v. dispose)

Reply:
Created a follow-up item for this [AB#2952](https://dev.azure.com/fluidframework/internal/_workitems/edit/2952)